client/asset/eth/eth.go

```diff
@@ -1089,6 +1089,23 @@ func (eth *ExchangeWallet) Refund(_, contract dex.Bytes, feeSuggestion uint64) (
 		return nil, fmt.Errorf("Refund: failed to decode contract: %w", err)
 	}
 
+	unlockFunds := func() {
+		eth.unlockFunds(eth.gasFeeLimit * dexeth.RefundGas(version))
+	}
+
+	swap, err := eth.node.swap(eth.ctx, secretHash, version)
+	if err != nil {
+		return nil, err
+	}
+	// It's possible the swap was refunded by someone else. In that case we
+	// cannot know the refunding tx hash.
+	if swap.State == dexeth.SSRefunded {
```

Comment:
BTW, I was sure the `Core`'s `isRefundable` checks would need something to account for this possibility, but it does not appear to matter.  Hopefully were not missing something.